Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+SwiftThunkPrinting.swift

```diff
@@ -0,0 +1,270 @@
+
+      try printGlobalSwiftThunkSources(&printer)
+
+      if let outputFile = try printer.writeContents(
+        outputDirectory: self.swiftOutputDirectory,
+        javaPackagePath: nil,
+        filename: moduleFilename
+      ) {
+        print("[swift-java] Generated: \(moduleFilenameBase.bold).swift (at \(outputFile))")
+        self.expectedOutputSwiftFiles.remove(moduleFilename)
+      }
+
+      for (_, ty) in self.analysis.importedTypes.sorted(by: { (lhs, rhs) in lhs.key < rhs.key }) {
+        let fileNameBase = "\(ty.swiftNominal.qualifiedName)+SwiftJava"
+        let filename = "\(fileNameBase).swift"
+        logger.info("Printing contents: \(filename)")
+
+        do {
+          try printNominalTypeThunks(&printer, ty)
+
+          if let outputFile = try printer.writeContents(
+            outputDirectory: self.swiftOutputDirectory,
+            javaPackagePath: nil,
+            filename: filename) {
+            print("[swift-java] Generated: \(fileNameBase.bold).swift (at \(outputFile))")
+            self.expectedOutputSwiftFiles.remove(filename)
+          }
+        } catch {
+          logger.warning("Failed to write to Swift thunks: \(filename)")
+        }
+      }
+    } catch {
+      logger.warning("Failed to write to Swift thunks: \(moduleFilename)")
+    }
+  }
+
+  private func printGlobalSwiftThunkSources(_ printer: inout CodePrinter) throws {
+    printHeader(&printer)
+
+    for decl in analysis.importedGlobalFuncs {
+      printSwiftFunctionThunk(&printer, decl)
+      printer.println()
+    }
+  }
+
+  private func printNominalTypeThunks(_ printer: inout CodePrinter, _ type: ImportedNominalType) throws {
+    printHeader(&printer)
+
+    for initializer in type.initializers {
+      printInitializerThunk(&printer, initializer)
+      printer.println()
+    }
+
+    for method in type.methods {
+      printSwiftFunctionThunk(&printer, method)
+      printer.println()
+    }
+  }
+
+  private func printInitializerThunk(_ printer: inout CodePrinter, _ decl: ImportedFunc) {
+    let translatedDecl = translatedDecl(for: decl)
+    // Initializers must have a parent
+    let typeName = translatedDecl.parentName!
+
+    printCDecl(
+      &printer,
+      javaMethodName: "allocatingInit",
+      parentName: translatedDecl.parentName,
+      parameters: translatedDecl.translatedFunctionSignature.parameters,
+      isStatic: true,
+      resultType: .long
+    ) { printer in
+      let downcallArguments = renderDowncallArguments(
+        swiftFunctionSignature: decl.functionSignature,
+        translatedFunctionSignature: translatedDecl.translatedFunctionSignature
+      )
+      // TODO: Throwing initializers
+      printer.print(
+        """
+        let selfPointer = UnsafeMutablePointer<\(typeName)>.allocate(capacity: 1)
+        selfPointer.initialize(to: \(typeName)(\(downcallArguments)))
+        return Int64(Int(bitPattern: selfPointer)).getJNIValue(in: environment)
```

Comment:
Yeah the int64 here is okey I think, and the generated signatures all just work on `long` which makes sense... If running on 32bit I think we'd likely do the same anyway, and just check at runtime values make sense maybe.

Reply:
I have a todo about thinking about 32-bit later! but I guess some Android devices can run 32-bit, so we probably have to think about it.